### PR TITLE
[PeerConnector] Fix logic error in PeerConnectorDiscovery

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin/P2P/PeerConnectorDiscovery.cs [hunk 1]
```diff
@@ -57,8 +57,16 @@ public override PeerAddress FindPeerToConnectTo()
 
             PeerAddress peer = null;
 
-            while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested && peerSelectionFailed < 50)
+            while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
             {
+                if (peerSelectionFailed > 50)
```
Comment (dangershony) [inline src/Stratis.Bitcoin/P2P/PeerConnectorDiscovery.cs]: I think we should put a wait here

Comment (fassadlr) [reply src/Stratis.Bitcoin/P2P/PeerConnectorDiscovery.cs]: Yeah it could be beneficial.... How long?

Comment (fassadlr) [reply src/Stratis.Bitcoin/P2P/PeerConnectorDiscovery.cs]: Maybe put it after each failed attempt?

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 2]
```diff
@@ -214,7 +214,10 @@ private Task ConnectAsync()
             {
                 PeerAddress peerAddress = this.FindPeerToConnectTo();
                 if (peerAddress == null)
+                {
+                    Task.Delay(TimeSpans.TenSeconds.Milliseconds);
```
Comment (dangershony) [inline src/Stratis.Bitcoin/P2P/PeerConnector.cs]: Task.Delay(TimeSpans.TenSeconds.Milliseconds).Wait(this.lifeteime.Cancalltaion)

**Pull request merged**